### PR TITLE
Release/1.2.31

### PR DESCRIPTION
**Pull request by AndyTWF**
Updates version numbers and changelog for the 1.2.31 release

### CODE REVIEW

**Review by sacOO7 — CHANGES_REQUESTED** [review]
Btw, you might like to merge https://github.com/ably/ably-java/pull/957 before we make a release. It's a small PR.

**Comment by sacOO7** [comment]
I believe https://github.com/ably/ably-flutter/issues/472 will be dependent on the fix made in this release.

**Review by sacOO7 — APPROVED** [review]
LGTM : )

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,23 @@
...
+## [1.2.31](https://github.com/ably/ably-java/tree/v1.2.31)
+
+[Full Changelog](https://github.com/ably/ably-java/compare/v1.2.30...v1.2.31)
+
+**Fixed bugs:**
+
+- Error code for channel attachment timed out [\#959](https://github.com/ably/ably-java/issues/959)
+- Error code for message decoding failure [\#958](https://github.com/ably/ably-java/issues/958)
+- Fix incremental backoff while reconnecting [\#954](https://github.com/ably/ably-java/issues/954)
+- Add missing clientOptions [\#956](https://github.com/ably/ably-java/issues/956)
```
Comment (owenpearson) [inline CHANGELOG.md:12]: tbh i think all of these issue titles are quite vague/non-obvious, could we editorialise them a bit?
```suggestion
- Update error code for channel attachment timed out [\#959](https://github.com/ably/ably-java/issues/959)
- Update error code for message decoding failure [\#958](https://github.com/ably/ably-java/issues/958)
- Fix incremental backoff while reconnecting [\#954](https://github.com/ably/ably-java/issues/954)
- Add `suspendedRetryTimeout` and `httpMaxRetryDuration`  [\#956](https://github.com/ably/ably-java/issues/956)
```

**Pull request merged**